design/one-pager-release-process.md

```diff
@@ -0,0 +1,172 @@
+# Release Process and Engineering
+
+## Background
+
+The current lifecycle process for a new version of the Crossplane software has gotten the project through the `v0.3` release, but also has a number of shortcomings.
+A summary of the current process that is very succinct, and skips many of the details, would look like the following:
+
+1. The team creates a [milestone](https://github.com/crossplaneio/crossplane/milestones) to represent the current/active release
+1. Issues are added to the milestone and taken through their lifecycle on the [project board](https://github.com/crossplaneio/crossplane/projects)
+1. The team continues to discuss the scope of the milestone and what issues should be included, through up front planning sessions, backlog grooming, and daily stand-ups
+1. When all issues for the milestone have been completed and merged into `master`, the official release process begins
+    1. The HEAD commit from `master` is tagged with the release version (e.g., [`v0.3.0`](https://github.com/crossplaneio/crossplane/releases/tag/v0.3.0))
+    and the release branch is created (e.g., [`release-0.3`](https://github.com/crossplaneio/crossplane/tree/release-0.3))
```

Comment:
I find release branches can be redundant to release tags. In the past we I have used a release branch until it was released (and tagged) and then scrapped the release branch.

The release branch could also act as a branch that holds the latest v0.3.x tag.

What's your take?

Reply:
I grep'd for `release-` and missed this section which explains the release branch.

https://github.com/crossplaneio/crossplane/pull/931/files#diff-f31a09f3f47ef35a6f5d5b83aa497071R78-R80 :+1: